### PR TITLE
Fix version getting, remove duplicated version file, fixes #405 

### LINKED ISSUE #405: Error when pip installing lstchain
I tried to install lstchain with pip but it complained of a versioning-related errors

```
$ pip install lstchain
Collecting lstchain
  Using cached lstchain-0.5.0.tar.gz (120 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/dmorcuende/.local/miniconda3/envs/py37/bin/python /home/dmorcuende/.local/miniconda3/envs/py37/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpd_zq6g82
       cwd: /tmp/pip-install-6krz4e5t/lstchain
  Complete output (16 lines):
  Traceback (most recent call last):
    File "/home/dmorcuende/.local/miniconda3/envs/py37/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 257, in <module>
      main()
    File "/home/dmorcuende/.local/miniconda3/envs/py37/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 240, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/dmorcuende/.local/miniconda3/envs/py37/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 91, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-i2aezbz7/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 148, in get_requires_for_build_wheel
      config_settings, requirements=['wheel'])
    File "/tmp/pip-build-env-i2aezbz7/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 128, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-i2aezbz7/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 143, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 10, in <module>
      from version import get_version, update_release_version  # noqa
  ModuleNotFoundError: No module named 'version'
  ----------------------------------------
ERROR: Command errored out with exit status 1: /home/dmorcuende/.local/miniconda3/envs/py37/bin/python /home/dmorcuende/.local/miniconda3/envs/py37/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpd_zq6g82 Check the logs for full command output.
```

### PR DESCRIPTION
**Pull request by MaxNoe**
While the the error that caused #405 was rather simple, namely missing `[]` brackets arround `pymodules='version.py'` should have been `pymodules=['version.py']`, while investigating this I found a solution to get rid of the duplicated version files that works with sdists, wheels and local installs. 

After merging this, we should tag as `v0.5.0.post1`, meaning it's the same version but with fixes unrelated to the code itself. See https://www.python.org/dev/peps/pep-0440/

### CODE REVIEW

File: lstchain/version.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 """
 from subprocess import check_output, CalledProcessError
 from os import path, name, devnull, environ, listdir
+from ast import literal_eval
 
 __all__ = ("get_version",)
```
Comment (rlopezcoto) [inline lstchain/version.py:34]: shouldn't this `__all__` contain all the functions defined in the file?

File: setup.py [hunk 2]
```diff
@@ -5,13 +5,18 @@
...
+# this is needed as lstchain/__init__.py imports dependencies
+# that might not be installed before setup runs, so we cannot import
+# lstchain.version
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), 'lstchain'))
 from version import get_version, update_release_version  # noqa
 
 
 update_release_version()
 version = get_version()
+print(version)
```
Comment (rlopezcoto) [inline setup.py:18]: a plane `print` here? 
maybe `print(f'lstchain version {version}')`

**Comment by rlopezcoto** [comment]
> After merging this, we should tag as v0.5.0.post1, meaning it's the same version but with fixes unrelated to the code itself. See https://www.python.org/dev/peps/pep-0440/

yes, thanks for the tip. After this is merged, I'll make the release. Please everybody, do not merge any other PRs in the meantime.

File: setup.py (hunk 2 above)
Comment (MaxNoe) [reply setup.py:18]: Debugging left over,  will remove

File: lstchain/version.py (hunk 1 above)
Comment (MaxNoe) [reply lstchain/version.py:34]: I think `get_version` was the only function meant to be part of the API, to be used in `__init__` for example. Actually this file is probably completely private, right? 

So `__all__` does not really have a relevance here.

**Pull request merged**